docs/how_to_write_custom_backend.rst

```diff
@@ -0,0 +1,193 @@
+        self.client.send_message({
+            'from': self._from,
+            'to': number,
+            'text': message,
+        })
+
+    def send_bulk_sms(self, numbers, message):
+        for number in numbers:
+            self.send_sms(self, number=number, message=message)
+
+How to create custom Sandbox Service
+------------------------------------
+
+The above steps will remain same if you wish to create a sandbox utility for your service. We'll create a new class with keeping above steps in mind. Apart from it, we will need to override a few more methods and tweak our ``__init__`` method a bit.
+
+1. Create a custom sandbox class for your service. Let's say ``NexmoSandboxBackend``. Again, it must inherit from ``BaseBackend``.
+
+2. Define your class's constructor same as above. Just fetch ``SANDBOX_TOKEN`` from the settings as well. Also, override ``send_sms`` and ``send_bulk_sms`` for your service same as above.
+
+.. code-block:: python
+
+    # Third Party Stuff
+    import nexmo
+    from phone_verify.backends.base import BaseBackend
+
+
+    class NexmoSandboxBackend(BaseBackend):
+
+        def __init__(self, **options):
+            super().__init__(**options)
+
+            # Lower case it just to be sure
+            options = {key.lower(): value for key, value in options.items()}
+            self._key = options.get("key", None)
+            self._secret = options.get("secret", None)
+            self._from = options.get("from", None)
+            self._token = options.get("sandbox_token", None) # Fetch sandbox token for your service.
+
+            # Create a Nexmo Client object
+            self.client = nexmo.Client(key=self._key, secret=self._secret)
+
+        def send_sms(self, number, message):
+            # Implement your service's send sms functionality
+            self.client.send_message({
+                'from': self._from,
+                'to': number,
+                'text': message,
+            })
+
+        def send_bulk_sms(self, numbers, message):
+            for number in numbers:
+                self.send_sms(self, number=number, message=message)
+
+3. Override ``generate_security_code`` as defined below:
+
+.. code-block:: python
+
+    ...
+
+    def generate_security_code(self):
+        """
+        Returns a fixed security code
+        """
+        return self._token
+
+4. Override ``validate_security_code`` as defined below:
+
+.. code-block:: python
+
+    ...
+
+    def generate_security_code(self):
+        """
+        Returns a fixed security code
+        """
+        return self._token
+
+    def validate_security_code(self, security_code, phone_number, session_token):
+        return self.SECURITY_CODE_VALID
+
+It must have ``security_code``, ``phone_number`` and ``session_token`` as its positional parameters.
+
+Now your Sandbox class is ready to be used. To use this class, give path to this class in your ``PHONE_VERIFY`` settings ``BACKEND`` key.
```

Comment:
In order to use this new custom backend class, it should be replaced in the ``BACKEND`` key under ``PHONE_VERIFICATION`` settings as shown below.